streams/src/commonTest/kotlin/com/mirego/trikot/streams/reactive/processors/SharedProcessorTests.kt

```diff
@@ -72,4 +74,22 @@ class SharedProcessorTests {
         assertEquals(1, executionCount)
         assertEquals("a", secondSubscriptionValueReceived)
     }
+
+    @Test
+    fun processorUnsubscribeFromParentOnError() {
+        val masterPublisher = MockPublisher("a")
+        val erroredPublisher = masterPublisher.map { throw StreamsProcessorException() }
+        val sharedPublisher = erroredPublisher.shared()
+        masterPublisher.subscribe(CancellableManager()) {}
+
+        var errorCount = 0
+        sharedPublisher.subscribe(CancellableManager(), {},
+            {
+                errorCount++
+                masterPublisher.value = "b"
```

Comment:
It's not clear where the validation that we are unsubscribed is 🤔

Reply:
I tried to make this clear yesterday but did not find a way, if we are not unsubscribed from parent, it throws an exception. I could also code my own masterPublisher to keep a number of subscriptions ?